### ISSUE TITLE
.tnz files from Tahoma2D not opening in Opentoonz

### ISSUE THREAD

**Issue by RodneyBaker** [tahoma2d/tahoma2d#879]
Note:  I closed my previous report #726 in favor of this new report.**
Much of the information is the same however.

**Situation producing incompatibility of .tnz files with Opentoonz:**
Currently .tnz files open fine in both Opentoonz and Tahoma2D.
.tnz scene created or edited in Tahoma2D however are not saved in a manner which allows those files to easily be opened in Opentoonz.  

A known workaround exists to manually allow T2D flavored .tnz files to open in Opentoonz.
(I'll post a link to that information if applicable but forget exact steps at the moment)

**Proposal**
Consider changing the scene files in Tahoma2D to a different file extension;  .t2d would seem to be appropriate.
This 'new' scene file format would be a derivative of the .tnz scene file format allowing T2D to move forward without specific concern for breaking access to .tnz scenes in other programs expecting compatibility with Opentoonz (as well as direct forks of the source code). 

Differences between the .t2d and .tnz formats can then be better monidtored for changes and documented to best address conversion needs performed programmatically or manually.

This would differentiate the files to prevent inadvertent atttempts to open incompatible files.

There may be other and better options.
This might be in the form of an exporter that specifically sets up files for use with .tnz compatiblity.
But even in this case the concern remains about .tnz files that make their way to programs that have no way of knowing about the loss of compatiblity.

Bottom line:
.tnz scenes created in Tahoma2D cannot be opened directly in Opentoonz.

Other thoughts?
 

**I forgot I submitted the earlier report

**Comment by manongjohn** [comment]
Sorry, I have no intentions of doing this.

Though I strive to ensure scenes created in T2D can open in OT, there is currently only 1 thing that I am aware of that can prevent that: Motion Paths.  Unfortunately the decision to add new Motion Path tags to the scene file was done and cannot really turn it back. 

What should be done is for someone to modify OT to not crash or ignore scene tags that are unknown and load whatever it can without canceling mid-loading of a scene as it does now. T2D actually needs this as well.

It is something I started working on for T2D but ran into some issues and set it aside until I had more time for it.

**Comment by RodneyBaker** [comment]
Thanks for the feedback.

We'll have to see what we as a community can do to assist.
This is something that will prevent adoption of T2D by some and frustrate Opentoonz users also as they encounter those files. 
 As such we'll eventually have to advance a solution.  The suggestion you make... for Opentoonz and T2D to ignore unrecognized tags... is a good one on many levels not just for this issue.

Knowing your thoughts on the effect Motion paths have on scene files does help in better troubleshooting issues that arise.

**Comment by manongjohn** [comment]
In order to maintain compatibility and seeing as T2D created the incompatibility, I will submit a fix to OT to either ignore the extra tags or read/store them as well as write out values (default them if missing) that T2D can use.

I just discovered that T2D can crash if you attempt to touch the Motion Path generated by OT in the schematic viewer so that needs to be fixed anyway.

**Comment by manongjohn** [comment]
PRs submitted. Closing.
